chatterbot/adapters/plugins/evaluate_mathematically.py

```diff
@@ -0,0 +1,183 @@
+from .plugin import PluginAdapter
+import re
+import os, json
+import decimal
+
+class EvaluateMathematically(PluginAdapter):
+
+    def should_answer(self, input_text):
+        """
+        Determines whether it is appropriate for this plugin
+        to respond to the user input.
+        """
+
+        response = self.process( input_text )
+
+        if response is False:
+            return False
+        else:
+            return True
+
+
+    def process(self, input_text):
+        """
+        Takes a statement string.
+        Returns the simplified statement string
+        with the mathematical terms "solved".
+        """
+
+        # Getting the mathematical terms within the input statement
+        expression = self.simplify_chunks( self.normalize( input_text ) )
+
+        # Returning important information
+        try:
+            expression += '= ' + str( eval( expression ) )
+
+            return expression
+        except:
+            return False
+
+
+    def simplify_chunks(self, input_text):
+        """
+        Separates the incoming text.
+        """
+
+        string = ''
+
+        for chunk in input_text.split( ' ' ):
+
+            is_chunk_integer = self.is_integer( chunk )
+
+            if is_chunk_integer is False:
+                is_chunk_float = self.is_float( chunk )
+
+                if is_chunk_float is False:
+                    is_chunk_operator = self.is_operator( chunk )
+
+                    if is_chunk_operator is False:
+                        continue
+                    else:
+                        string += str( is_chunk_operator ) + ' '
+                else:
+                    string += str( is_chunk_float ) + ' '
+            else:
+                string += str( is_chunk_integer ) + ' '
+
+        return string
+
+
+    def is_float(self, string):
+        """
+        If the string is a float, returns
+        the float of the string. Otherwise,
+        it returns False.
+        """
+
+        try:
+            return decimal.Decimal(string)
+        except decimal.DecimalException:
+            return False
+
+
+    def is_integer(self, string):
+        """
+        If the string is an integer, returns
+        the int of the string. Otherwise,
+        it returns False.
+        """
+
+        try:
+            return int( string )
+        except:
+            return False
+
+
+    def is_operator(self, string):
+        """
+        If the string is an operator, returns
+        said operator. Otherwise, it returns
+        false.
+        """
+
+        if string in "+-/*^()":
+            return string
+        else:
+            return False
+
+
+    def normalize(self, string):
+        """
+        Normalizes input text, reducing errors
+        and improper calculations.
+        """
+
+        # Setting all words to lowercase
+        string = string.lower()
+
+        # Removing punctuation
+        if string.endswith( ('.', '!', '?', ':', ';' ) ):
```

Comment:
If might be a good idea to handle any non-alphanumeric character rather than just this specific set.
`if not string[-1].isalnum():` would check if the last character in the string is not an alphanumeric character.